examples/dom/WebReconciler.re

```diff
@@ -0,0 +1,229 @@
+/**
+  * This is an implementation of a reconciler for DOM elements via js_of_ocaml :
+  * http://ocsigen.org/js_of_ocaml/3.1.0/api/Dom_html
+  *
+  * This is just an example but you could use this to create interesting
+  * CLI apps, with a react-like functional API!
+*/
+
+exception InvalidNodePrimitiveMatchInUpdateInstance;
+
+let str = string_of_int;
+
+module Dom = Js_of_ocaml.Dom;
+module Dom_html = Js_of_ocaml.Dom_html;
+module Js = Js_of_ocaml.Js;
+
+/*
+   Step 1: Define the reconciler template
+ */
+module Reconciler = {
+  type hostElement = Js.t(Dom_html.element);
+  type node = Js.t(Dom_html.element);
+
+  let onStale: EventDom.t(unit) = EventDom.create();
+
+  let insertNode = (~parent: node, ~child: node, ~position as _) => {
+    Dom.appendChild(parent, child);
+    parent;
+  };
+
+  let deleteNode = (~parent: node, ~child: node) => {
+    Dom.removeChild(parent, child);
+    parent;
+  };
+
+  let moveNode = (~parent, ~child as _, ~from as _, ~to_ as _) => {
+    parent;
+  };
+
+  let markAsStale = () => EventDom.dispatch(onStale, ());
```

Comment:
really cool.

Reply:
All credit goes to @bryphe. Not only for this specific part, but for more than half of this PR 😄